test/ipfs.test.ts

```diff
@@ -108,6 +109,22 @@ describe("IPFS Uploads", async () => {
     const cid = await sdk.getStorage().uploadBatch(sampleObjects);
   });
 
+  it("should upload files with filenames correctly", async () => {
+    const sampleObjects: FileOrBufferWithNames[] = [
+      {
+        file: readFileSync("test/test.mp4"),
+        name: "test2.mp4",
+      },
+      { file: readFileSync("test/test.mp4"), name: "test3.mp4" },
+      {
+        file: readFileSync("test/test.mp4"),
+        name: "test1.mp4",
+      },
+    ];
+    const cid = await sdk.getStorage().uploadBatchWithFileNames(sampleObjects);
+    console.log(cid);
+  });
+
```

Comment:
Two things we need in this test suite in order to merge:

1. [ ] A test for the `uploadBatch` that:
    - [ ] Calls upload with `uploadFileStartNumber` = to something > 0, then ensures that fetching all files at valid indexes works (like `{cid}/10`, `{cid}/11` ...)
    - [ ] Calls upload with `upoadFileStartNumber` = to 0, and checks all fetched files
2. [ ] Update the `should upload files with filenames correctly` test to actually fetch the files using their file names and ensures correct content
    - [ ] Fetching a `{cid}/test1.mp4` should work